homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,271 @@
+"""Config flow to configure the Nextbus integration."""
+from collections.abc import Iterable
+import logging
+
+from py_nextbus import NextBusClient
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import (
+    CONF_AGENCY,
+    CONF_AGENCY_NAME,
+    CONF_ROUTE,
+    CONF_ROUTE_NAME,
+    CONF_STOP,
+    CONF_STOP_NAME,
+    DOMAIN,
+)
+from .util import UserConfig, invert_dict
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _agency_schema(agency_names: Iterable[str]) -> vol.Schema:
+    return vol.Schema({vol.Required(CONF_AGENCY_NAME): vol.In(sorted(agency_names))})
+
+
+def _route_schema(route_names: Iterable[str]) -> vol.Schema:
+    return vol.Schema({vol.Required(CONF_ROUTE_NAME): vol.In(sorted(route_names))})
+
+
+def _stop_schema(stop_names: Iterable[str]) -> vol.Schema:
+    return vol.Schema({vol.Required(CONF_STOP_NAME): vol.In(sorted(stop_names))})
```

Comment:
Please look at `SelectSelector` and `SelectSelectorConfig` instead.
Also, there is no need to create a function that is only used once.